test/scripts/get-missing-cherry-picks.test.ts

```diff
@@ -0,0 +1,231 @@
+      data: {status: 'diverged'},
+    } as CompareResponse);
+    when(
+      spiedRepos.compareCommitsWithBasehead(
+        deepEqual({
+          owner: 'ampproject',
+          repo: 'amphtml',
+          basehead: '2202090001000...b0fd442d5abc691a3d6eea076eccb83525258b7b',
+        })
+      )
+    ).thenResolve({
+      status: 200,
+      data: {status: 'ahead'},
+    } as CompareResponse);
+
+    const releases = new Set([
+      '2202060001000', // lts
+      '2202070001002', // stable
+      '2202080001000', // beta-traffic
+    ]);
+    const commits = await getMissingCommits(octokit, '2202090001000', releases);
+    assert(commits.length == 2);
+    assert(
+      [
+        '725595ac41082b0f9db3103143b04070b0fa7a9e',
+        'b0fd442d5abc691a3d6eea076eccb83525258b7b',
+      ].every((c) => commits.includes(c))
+    );
+  });
+
+  it('stable and lts have the same cherry-pick that is not in beta-opt-in', async () => {
+    const compareReleasesResponse = {
+      status: 200,
+      data: {
+        commits: [
+          {
+            commit: {
+              message: `\
+              First PR to fix bug (#29041)
+              
+              * update worker-dom
+              * yarn lock
+              
+              (cherry picked from commit 725595ac41082b0f9db3103143b04070b0fa7a9e)`,
+            },
+          },
+        ],
+      },
+    } as CompareResponse;
+    when(
+      spiedRepos.compareCommitsWithBasehead(
+        deepEqual({
+          owner: 'ampproject',
+          repo: 'amphtml',
+          basehead: '2202060001000...2202060001001',
+        })
+      )
+    ).thenResolve(compareReleasesResponse);
+    when(
+      spiedRepos.compareCommitsWithBasehead(
+        deepEqual({
+          owner: 'ampproject',
+          repo: 'amphtml',
+          basehead: '2202070001000...2202070001001',
+        })
+      )
+    ).thenResolve(compareReleasesResponse);
+    when(
+      spiedRepos.compareCommitsWithBasehead(
+        deepEqual({
+          owner: 'ampproject',
+          repo: 'amphtml',
+          basehead: '2202090001000...725595ac41082b0f9db3103143b04070b0fa7a9e',
+        })
+      )
+    ).thenResolve({
+      status: 200,
+      data: {status: 'diverged'},
+    } as CompareResponse);
+
+    const releases = new Set([
+      '2202060001001', // lts
+      '2202070001001', // stable
+      '2202080001000', // beta-traffic
+    ]);
+    const commits = await getMissingCommits(octokit, '2202090001000', releases);
+    assert(commits.length == 1);
+    assert(commits[0] == '725595ac41082b0f9db3103143b04070b0fa7a9e');
+  });
+});
+
+describe('cases where cherry-picks are not missing', () => {
+  const octokit = new Octokit();
+  const spiedRepos = spy(octokit.rest.repos);
+  type CompareResponse = GetResponseTypeFromEndpointMethod<
+    typeof octokit.rest.repos.compareCommitsWithBasehead
+  >;
+
+  afterEach(() => {
+    resetCalls(spiedRepos);
+  });
+
+  after(() => {
+    reset(spiedRepos);
+  });
```

Comment:
Does mocha support nested describes? If so, initialize the consts and add the after/afterEach in a wrapping `describe` so you wouldn't have to repeat the setup